lib/pundit.rb

```diff
@@ -88,7 +88,8 @@ def policy_scope(user, scope)
     # @raise [NotDefinedError] if the policy scope cannot be found
     # @return [Scope{#resolve}] instance of scope class which can resolve to a scope
     def policy_scope!(user, scope)
-      PolicyFinder.new(scope).scope!.new(user, scope).resolve
+      model = scope.is_a?(Array) ? scope.last : scope
```

Comment:
It's very unfortunate behavior.  I think the only option might be to check `#model_name`, or drop support for Rails 3 (not my personal favorite choice, since I still maintain a Rails 3 application, but it would be understandable since even Rails doesn't support Rails 3 anymore).